app/controllers/CoursesController.php

```diff
@@ -309,24 +309,38 @@ public function search($level, $year, $search_type = '', $search_string = '')
 			return Flight::notFound();
 		}
 
-		switch($level){
-			case 'postgraduate':
+		// get all the years on offer, most recent first
+		$years_data = static::$pp->make_request("$level/year");
+		$years = array_reverse($years_data->years);
```

Comment:
You'll probably want to explicitly order the years as the one that comes first isnt always the current year.